CARDS-1236: Test runMode should provide a mock NCR service which will allow the NCRNote component to be tested with hard-coded text annotations

IntegralProgrammer opened the pull request:
This PR creates a servlet at `/ncr/annotate` which provides a mock NeuralCR service so that the NCRNote front-end component can be tested with hard-coded text annotations thus not requiring the CARDS test instance to be run in a Docker Compose environment that contains a NeuralCR container.

To test:
1. Build this branch
2. Start CARDS with the `test` runMode enabled
3. Install the HP vocabulary from BioPortal
4. Create a new _Concept Recognition Test_ Form
5. In the _Notes_ section for the _Co-morbidities_ question, the following texts (ensure that they are entered _exactly_ as shown) should be annotated as expected based on the hard-coded annotations provided by this PR:
  - `The patient has heart disease and diabetes`
  - `The patient has renal cancer and myopia`
  - `The patient has experienced chest pain and shortness of breath`
 
Please note that this mock NCR service will be used for all NCR annotations whenever the `test` runMode is enabled and a reverse proxy is not being used. This should not be a concern as in production environments the `test` runMode will not be enabled and a reverse proxy will be present to route everything under `/ncr/` to a real NCR container.

IntegralProgrammer committed:
CARDS-1236: Test runMode should provide a mock NCR service which will allow the NCRNote component to be tested with hard-coded text annotations

Create a mock NCR servlet at /ncr/annotate

sashaandjic reviewed (approved):
tested
![image](https://user-images.githubusercontent.com/23243244/127121916-6d9c648d-7d01-46c6-822c-eb716d0a579e.png)
![image](https://user-images.githubusercontent.com/23243244/127121943-f2412d66-c03a-4fa8-a0cd-5b3da146d261.png)
![image](https://user-images.githubusercontent.com/23243244/127121973-ec3286ab-c61d-4a34-8587-1fcddc9efc9b.png)

sdumitriu commented on test-resources/src/main/resources/SLING-INF/content/MockNCR/annotate.json:
```diff
@@ -0,0 +1,3 @@
+{
```
Is this needed? As far as I can see, just the new servlet is needed.

sdumitriu committed:
CARDS-1236: Test runMode should provide a mock NCR service which will allow the NCRNote component to be tested with hard-coded text annotations
Removed unneeded node

sdumitriu committed:
CARDS-1236: Test runMode should provide a mock NCR service which will allow the NCRNote component to be tested with hard-coded text annotations
More decriptive questionnaire name, and add a description of the test to perform

sashaandjic reviewed (approved):
re-tesed
